### PR TITLE
Fixes?

### PR DESCRIPTION
**Pull request by mschauer**

### CODE REVIEW

File: example/phylo/phylo2.jl [hunk 1]
```diff
@@ -86,7 +86,8 @@ S = "(A:0.1,B:0.2,(C:0.3,D:0.4)E:0.5)F;"
...
 
-tree = Tree(S20)
+tree = Tree(S50)
+@test sort(setdiff(1:length(tree.Par), tree.Par)) == sort(tree.lids)
```
Comment (mschauer) [inline example/phylo/phylo2.jl:90]: Let's check this (all vertices except leafs are parents).

File: example/phylo/phylo2.jl [hunk 2]
```diff
@@ -173,30 +174,63 @@ function bwfiltertree!(Q, tree::Tree, θlin, dt0)
...
+  NoiseWrapper(source′,reset=reset,reverse=reverse,indx=indx)
+end
 
 ## simulate conditioned process using messages
 
-function fwguidtree!(messages, X, guidedsegs, ll, tree::Tree, f, g, θ, rho)
+function fwguidtree!(messages, X, guidedsegs, noisesegs, ll, tree::Tree, f, g, θ, rho)
     T = tree.T
+    ll .= -1
+    ll[1] = 0.0
```
Comment (mschauer) [inline example/phylo/phylo2.jl:207]: Make sure that `ll[1] = 0`

File: example/phylo/phylo2.jl [hunk 3]
```diff
@@ -173,30 +174,63 @@ function bwfiltertree!(Q, tree::Tree, θlin, dt0)
...
+  #NoiseGrid(t, Vector.(W))
+  NoiseWrapper(source′,reset=reset,reverse=reverse,indx=indx)
+end
 
 ## simulate conditioned process using messages
 
-function fwguidtree!(messages, X, guidedsegs, ll, tree::Tree, f, g, θ, rho)
+function fwguidtree!(messages, X, guidedsegs, noisesegs, ll, tree::Tree, f, g, θ, rho)
     T = tree.T
+    ll .= -1
```
Comment (mschauer) [inline example/phylo/phylo2.jl:206]: putting sentinels here

File: example/phylo/phylo2.jl [hunk 4]
```diff
@@ -173,30 +174,63 @@ function bwfiltertree!(Q, tree::Tree, θlin, dt0)
...
+    ll[1] = 0.0
+    if isnothing(noisesegs)
+        noisesegs = guidedsegs
+    end
     for i in eachindex(T)
         i == 1 && continue  # skip root-node (has no parent)
         i′ = tree.Par[i]
+        @assert i′ < i
+    #    @show i′, i
+        @assert ll[i′] != -1
```
Comment (mschauer) [inline example/phylo/phylo2.jl:216]: Make sure parent has already changed `-1` into something meaningful

File: example/phylo/phylo2.jl [hunk 5]
```diff
@@ -173,30 +174,63 @@ function bwfiltertree!(Q, tree::Tree, θlin, dt0)
...
     T = tree.T
+    ll .= -1
+    ll[1] = 0.0
+    if isnothing(noisesegs)
+        noisesegs = guidedsegs
+    end
     for i in eachindex(T)
         i == 1 && continue  # skip root-node (has no parent)
         i′ = tree.Par[i]
+        @assert i′ < i
```
Comment (mschauer) [inline example/phylo/phylo2.jl:214]: i′ < i this implies that parent edge already computed

File: example/phylo/phylo2.jl (hunk 4 above)
Comment (fmeulen) [reply example/phylo/phylo2.jl:216]: What about setting `ll[1] = 0.0` in `fwguidtree!`?

Comment (mschauer) [reply example/phylo/phylo2.jl:216]: I just make sure the field does not contain `-1` which would indicate a mistake or bad luck

**Pull request closed without merging**